ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -102,8 +102,42 @@ gcp_connection::delete_object(const std::string &object_key)
 
 // Retrieves an object from the google cloud bucket.
 int
-gcp_connection::get_object(const std::string &object_key, const std::string &path) const
+gcp_connection::read_object(const std::string &object_key, int64_t offset, size_t len, void *buf)
 {
+    if (offset < 0) {
+        std::cerr << "Invalid argument, cannot have offset less than zero." << std::endl;
+        return -1;
+    }
+
+    bool exists = false;
+    size_t object_size;
+
+    if (object_exists(object_key, exists, object_size) == -1) {
```

Comment:
I suggest checking against success here rather than evaluating against a specific error return value. There's only one value returned for success but can be any number of error status values.
```
if (object_exists(object_key, exists, object_size) != 0) { ... }
```

Reply:
Updated here: 3bbbef09be494b5ea7334358a0654bc050b9def8